core/zms/src/main/rdl/Role.rdli

```diff
@@ -129,6 +129,20 @@ resource DomainRoleMembers GET "/domain/{domainName}/member" {
     }
 }
 
+// Fetch all the roles across domains by either calling or specified principal
+resource DomainRoleMember GET "/all_roles?principal={principal}&domain={domainName}" (name=GetAllRoles) {
```

Comment:
for the uri should be consistent with others and use the first component as the object type, e.g. /domain, /resource,e tc.

we should just call it /role instead of /all_roles and instead of getAllRoles, let's call the API as getPrincipalRoles

Reply:
Done